provider/core/src/lib.rs

```diff
@@ -168,6 +168,7 @@ pub mod prelude {
     pub use crate::data_provider::DataResponseMetadata;
     pub use crate::data_provider::DynProvider;
     pub use crate::data_provider::ResourceProvider;
+    pub use crate::data_provider::RcWrap;
```

Comment:
Suggestion: If we add this to the prelude, it should have a more unique name. Alternatively, just move it to the top level.